target/linux/generic/hack-5.15/290-net-dsa-mv88e6xxx-depend-on-PTP-conditionally.patch

```diff
@@ -0,0 +1,49 @@
+From 65e2a0b2a5f93011dde13907720c25741194fa75 Mon Sep 17 00:00:00 2001
+From: "Johnny S. Lee" <foss@jsl.io>
+Date: Thu, 15 Dec 2022 17:49:04 +0800
+Subject: [PATCH] net: dsa: mv88e6xxx: depend on PTP conditionally
+
+PTP hardware timestamping related objects are not linked when PTP support
+for MV88E6xxx is disabled, therefore NET_DSA_MV88E6XXX following the same
+state of PTP_1588_CLOCK_OPTIONAL should be prevented when
+NET_DSA_MV88E6XXX_PTP=n
+
+Fixes: e5f31552674e ("ethernet: fix PTP_1588_CLOCK dependencies")
+Signed-off-by: Johnny S. Lee <foss@jsl.io>
+---
+ drivers/net/dsa/mv88e6xxx/Kconfig | 17 +++++++++++++++--
+ 1 file changed, 15 insertions(+), 2 deletions(-)
+
+--- a/drivers/net/dsa/mv88e6xxx/Kconfig
++++ b/drivers/net/dsa/mv88e6xxx/Kconfig
+@@ -2,7 +2,7 @@
+ config NET_DSA_MV88E6XXX
+ 	tristate "Marvell 88E6xxx Ethernet switch fabric support"
+ 	depends on NET_DSA
+-	depends on PTP_1588_CLOCK_OPTIONAL
++	depends on NET_DSA_MV88E6XXX_PTP_OPTIONAL
+ 	select IRQ_DOMAIN
+ 	select NET_DSA_TAG_EDSA
+ 	select NET_DSA_TAG_DSA
+@@ -13,7 +13,20 @@ config NET_DSA_MV88E6XXX
+ config NET_DSA_MV88E6XXX_PTP
+ 	bool "PTP support for Marvell 88E6xxx"
+ 	default n
+-	depends on NET_DSA_MV88E6XXX && PTP_1588_CLOCK
++	depends on PTP_1588_CLOCK
++	select NET_DSA_MV88E6XXX
```

Comment:
and replace this with

```suggestion
-	depends on NET_DSA_MV88E6XXX && PTP_1588_CLOCK
+	depends on (PTP_1588_CLOCK = y && NET_DSA_MV88E6XXX = y) || \
+		   (PTP_1588_CLOCK && NET_DSA_MV88E6XXX = m)
```

like the [Marvell armada ethernet](https://github.com/torvalds/linux/blob/master/drivers/net/ethernet/marvell/Kconfig#L96) driver does.

This then should even be upstreamable.

Reply:
Thanks. This is better. Patch revised.